staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go

```diff
@@ -592,6 +592,19 @@ func (q *Quantity) Sub(y Quantity) {
 	q.ToDec().d.Dec.Sub(q.d.Dec, y.AsDec())
 }
 
+// Mul multiplies the provided y quantity to the current value. If the current value is zero,
+// the format of the quantity will be updated to the format of y.
+func (q *Quantity) Mul(y Quantity) {
```

Comment:
I strongly recommend multiplying by a scalar instead (i.e., just a plain int or int64).

Quantities are implicitly typed (either milliCPU or RAM) and multiplying them together gives units like squared milliCPU (which would be microCPU) or squared bytes. For your use case I think multiplying by a scalar makes much more sense. The implementation will be simpler too.

The actual invariant you need to test / prove is that repeated addition gives the same answer as the multiplication. The point of this type is that there isn't floating point error accumulation.

Reply:
I think the signature of this function needs to be able to visibly fail on overflow.

Reply:
or if not overflow, it could have a return value indicating when the result is inexact (e.g. because it had to change scale to avoid overflow and repeated addition might get a different final answer).

Reply:
Thank you for some suggestions!

> I strongly recommend multiplying by a scalar instead (i.e., just a plain int or int64).

It sounds much more reasonable to me. 

I will change the implementation so that this function has an arg typed int64 since I don't have any reasons to want to use a Quantity instead of a scaler.

IIUC, I can address your other comments once I change this implementation as you suggested (Quantity -> scalar).